api/db-cdr/generate-cdr/make-bq-dataset-linking.sh

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+
+# This generates the big query de-normalized tables for dataset builder.
+
+set -xeuo pipefail
+IFS=$'\n\t'
+
+
+# get options
+
+# --cdr=cdr_version ... *optional
+USAGE="./generate-clousql-cdr/make-bq-denormalized-dataset.sh --bq-project <PROJECT> --bq-dataset <DATASET>"
+
+while [ $# -gt 0 ]; do
+  echo "1 is $1"
+  case "$1" in
+    --bq-project) BQ_PROJECT=$2; shift 2;;
+    --bq-dataset) BQ_DATASET=$2; shift 2;;
+    -- ) shift; break ;;
+    * ) break ;;
+  esac
+done
+
+
+if [ -z "${BQ_PROJECT}" ]
+then
+  echo "Usage: $USAGE"
+  exit 1
+fi
+
+if [ -z "${BQ_DATASET}" ]
+then
+  echo "Usage: $USAGE"
+  exit 1
+fi
+
+
+# Check that bq_dataset exists and exit if not
+datasets=$(bq --project=$BQ_PROJECT ls)
+if [ -z "$datasets" ]
+then
+  echo "$BQ_PROJECT.$BQ_DATASET does not exist. Please specify a valid project and dataset."
+  exit 1
+fi
+if [[ $datasets =~ .*$BQ_DATASET.* ]]; then
+  echo "$BQ_PROJECT.$BQ_DATASET exists. Good. Carrying on."
+else
+  echo "$BQ_PROJECT.$BQ_DATASET does not exist. Please specify a valid project and dataset."
+  exit 1
+fi
+
+
+################################################
+# CREATE LINKING TABLE
+################################################
+echo "CREATE TABLE - ds_linking"
+bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
+"CREATE OR REPLACE TABLE \`$BQ_PROJECT.$BQ_DATASET.ds_linking\`
+(
+    DENORMALIZED_NAME               STRING,
+    OMOP_SQL                        STRING,
+    JOIN_VALUE                      STRING,
+    DOMAIN                          STRING,
+)"
+
+################################################
+# INSERT DATA
+################################################
+echo "ds_linking - inserting conditions data"
+
+bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
+"INSERT INTO \`$BQ_PROJECT.$BQ_DATASET.ds_linking\`
+VALUES
+    ('PERSON_ID', 'a.PERSON_ID', 'from \`$BQ_PROJECT.$BQ_DATASET.condition_occurrence\` a', 'Condition'),
+    ('CONDITION_CONCEPT_ID', 'a.CONDITION_CONCEPT_ID', 'from \`$BQ_PROJECT.$BQ_DATASET.condition_occurrence\` a', 'Condition'),
```

Comment:
IIUC, this table will be responsible for generating the user-facing SQL for a dataset – correct?

I don't think we should expect users to understand the reasoning behind these table names "a", "c1", "c2", etc.

If possible, we should switch to using meaningful (but concise) nicknames for these intermediate tables?

For example, a first pass on this table:

a -- > c_occurrence
c1 --> c_concept
c2 --> c_type
c3 --> visit
c4 --> c_source
c5 --> c_status

This would immediately make the resulting SQL much more interpetable to someone reading this for the first time. (And also easier for a developer to understand what's going on here.)

Reply:
Will change. That works for me.